pkg/cmd/dockerregistry/dockerregistry_test.go

```diff
@@ -0,0 +1,102 @@
+package dockerregistry
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/docker/distribution/configuration"
+)
+
+func TestDefaultMiddleware(t *testing.T) {
+	checks := []struct {
+		title, input, expect string
+	}{
+		{
+			title: "miss all middlewares",
+			input: `
+version: 0.1
+storage:
+  inmemory: {}
+`,
+			expect: `
+version: 0.1
+storage:
+  inmemory: {}
+middleware:
+  registry:
+    - name: openshift
+  repository:
+    - name: openshift
+  storage:
+    - name: openshift
+`,
+		},
+		{
+			title: "miss some middlewares",
+			input: `
+version: 0.1
+storage:
+  inmemory: {}
+middleware:
+  registry:
+    - name: openshift
+`,
+			expect: `
+version: 0.1
+storage:
+  inmemory: {}
+middleware:
+  registry:
+    - name: openshift
+  repository:
+    - name: openshift
+  storage:
+    - name: openshift
+`,
+		},
+		{
+			title: "all middlewares are in place",
+			input: `
+version: 0.1
+storage:
+  inmemory: {}
+middleware:
+  registry:
+    - name: openshift
+  repository:
+    - name: openshift
+  storage:
+    - name: openshift
+`,
+			expect: `
+version: 0.1
+storage:
+  inmemory: {}
+middleware:
+  registry:
+    - name: openshift
+  repository:
+    - name: openshift
+  storage:
+    - name: openshift
+`,
+		},
```

Comment:
add test cases with the 1.1, 1.2, and 1.3 config data. that'll also make sure we don't overwrite an existing middleware stanza